blueocean-rest/src/main/java/io/jenkins/blueocean/rest/ContainerFilter.java

```diff
@@ -0,0 +1,80 @@
+package io.jenkins.blueocean.rest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+
+import com.google.common.base.Predicate;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Item;
+
+/**
+ * Simple extension point to allow filtering item types by a specific key
+ * This can be used anywhere a list of items are returned, and it will examine
+ * the current request for a ?filter=filter1(param:value),filter2 parameter,
+ * okay the last bit is a future addition once/if OmniSearch and this are consolidated
+ */
+public abstract class ContainerFilter implements ExtensionPoint {
```

Comment:
@kzantow yes this is fine as this is indeed single instance jenkins specific extension. We are keeping such extension points (e.g. BluePipelineFactory, BlueRunFactory) inside rest-impl.